### PR TITLE
Change User Transformations applying order in MO

### PR DESCRIPTION
**Pull request by daniil-lyakhov**
### Details:
Ngraph pruning transformation expects IR model with all mo optimization passes applied. In case when `--transfrom=Pruning` cli used, pruning transformation is receiving model before mock and fake quantize optimization passes applied, which leads to incorrect pruning transformation work. 

Suggested order of transformations allows to apply pruning transformation correctly.

### Ticket:
78682

### CODE REVIEW

**Review by GlebKazantaev — APPROVED** [review]

**Review by alexsu52 — APPROVED** [review]

File: tools/mo/openvino/tools/mo/back/offline_transformations.py [hunk 1]
```diff
@@ -94,8 +94,8 @@ def read_model(path_to_xml):
     if 'scale' in argv:
         argv.scale = scale
 
-    apply_user_transformations(func, parse_transform(argv.transform))
     apply_moc_transformations(func)
+    apply_user_transformations(func, parse_transform(argv.transform))
```
[the remark this answers is not in the document]
Comment (GlebKazantaev) [reply tools/mo/openvino/tools/mo/back/offline_transformations.py:98]: It won't impact LowLatency as in MOC we apply only fusions that only could help to perform LowLatency, Pruning and other user-defined transformations.

**Comment by daniil-lyakhov** [comment]
Convert job 1074 reveals some model can't be converted with pruning turned on by default with such order of transformation. Looks like some models can't be pruned before fp16 compression. In latest changes pruning transformation is applying just before serialization, so, now only fully transformed models will be fed to pruning transformation.

**Comment by rkazants** [comment]
@daniil-lyakhov, @GlebKazantaev, what kind of testing was performed against this change? Not sure that we can rely on the pre-commit. Did you check this change for other user transformations?

**Comment by GlebKazantaev** [comment]
> @daniil-lyakhov, @GlebKazantaev, what kind of testing was performed against this change? Not sure that we can rely on the pre-commit. Did you check this change for other user transformations?

Pruning transformation was tests by @daniil-lyakhov. LowLatency unfortunately is not tested in our CI by some e2e and covered only with unit tests which enough since LL doesn't depend on graph structure and relies only on inputs and outputs of models that we do not change during MOC. So for LL it doesn't matter when it will be applied.

**Comment by mvafin** [comment]
@daniil-lyakhov Why are we moving user transformations step after FP16 compression? It wasn't expected. Converts inserted by FP16 compression transformation weren't marked as "disable_constant_folding" since it was last before serialization. If we want to add transformation after it we need to do that marking or otherwise call to ConstantFolding transformation will fold those Converts.

**Review by rkazants — APPROVED** [review]
I am fine with this change if @mvafin and @itikhono are fine as well

**Comment by itikhono** [comment]
Looks like moc and legacy moc transformations don't affect TensorIterator and tensor names of Parameter/Result nodes, so LowLatency and MakeStateful transformations should not be affected either.
LGTM

**Review by itikhono — APPROVED** [review]

**Review by mvafin — CHANGES_REQUESTED** [review]
Currently, it is dangerous to move user transformations after FP16 compression. See my comment earlier.

**Comment by daniil-lyakhov** [comment]
> Currently, it is dangerous to move user transformations after FP16 compression. See my comment earlier.

Looks like problem with pruning transformation isn't related to FP16 compression, it's sporadic issue inside pruning transformation. Have changed that, sorry for inconvenience

**Review by mvafin — APPROVED** [review]

**Comment by daniil-lyakhov** [comment]
@GlebKazantaev, please merge if there are no comments

**Pull request merged**